WeatherApp/WeatherApp.Webpage/Startup.cs

```diff
@@ -32,11 +33,14 @@ public void ConfigureServices(IServiceCollection services)
                                   });
             });
 
-            services.AddTransient<ICurrentWeatherService, CurrentWeatherService>();
-            services.AddTransient<WeatherForecastService, WeatherForecastService>();
-            services.AddTransient<AutocompleteService, AutocompleteService>();
-            services.AddSingleton<FavoriteContext, FavoriteContext>();
+            services.AddSingleton<ICurrentWeatherService, CurrentWeatherService>();
+            services.AddSingleton<IWeatherForecastService, WeatherForecastService>();
+            services.AddSingleton<IAutocompleteService, AutocompleteService>();
+            services.AddSingleton<IFavoritesRepository, InMemoryFavoritesRepository>();

```

Comment:
Define your Repository in the service collection as Scoped because EF Core manages DBContext as scoped service!
See further info on this topic this article: <https://ardalis.com/avoid-wrapping-dbcontext-in-using/>

Reply:
In this case, the application doesn't use EF core for those in-memory repositories, but simple in-memory data structures. Is this ok?

Reply:
In this case ok. But don't forget to change it when you switch to EF repository implementation.

Reply:
Sure, thank you!